src/nvim/CMakeLists.txt

```diff
@@ -401,6 +406,12 @@ if(JEMALLOC_FOUND)
   list(APPEND NVIM_EXEC_LINK_LIBRARIES ${JEMALLOC_LIBRARIES})
 endif()
 
+if(WIN32 AND NVIM_BUILD_STATIC)
+  list(REMOVE_DUPLICATES NVIM_EXEC_LINK_LIBRARIES)
+  # Static stdc++ runtime library must keep tail for static link 
+  list(APPEND NVIM_EXEC_LINK_LIBRARIES -static-libgcc -static-libstdc++ -Wl,-Bstatic -lstdc++)
```

Comment:
Maybe it's worth doing REMOVE_DUPLICATES after this line, then append `-lstdc++`. Like this:

```
list(APPEND NVIM_EXEC_LINK_LIBRARIES -static-libgcc -static-libstdc++ -Wl,-Bstatic)
list(REMOVE_DUPLICATES NVIM_EXEC_LINK_LIBRARIES)
list(APPEND NVIM_EXEC_LINK_LIBRARIES -lstdc++)
```

Also, should there be a space in `-Wl,-Bstatic`? Like this: `-Wl, -Bstatic`

Reply:
> Also, should there be a space in -Wl,-Bstatic? Like this: -Wl, -Bstatic (I don't know, just asking)

According to ld manual,
```
---if the linker is being invoked indirectly, via a compiler driver (e.g. gcc) then all the linker command line options should be prefixed by -Wl,(or whatever is appropriate for the particular compiler driver) like this:

gcc -Wl,--start-group foo.o bar.o -Wl,--end-group
```
So it should not separate by a space.

Reply:
Done, followed you advice, but minor change.